cluster/kubernetes/kubernetes.go

```diff
@@ -315,20 +313,29 @@ func (c *Cluster) PublicSSHKey(regenerate bool) (ssh.PublicKey, error) {
 // instance, in which case it returns a list containing the namespaces from the whitelist
 // that exist in the cluster.
 func (c *Cluster) getAllowedNamespaces() ([]apiv1.Namespace, error) {
-	nsList := []apiv1.Namespace{}
+	if len(c.nsWhitelist) > 0 {
+		nsList := []apiv1.Namespace{}
+		for _, name := range c.nsWhitelist {
+			ns, err := c.client.CoreV1().Namespaces().Get(name, meta_v1.GetOptions{})
+			switch {
+			case err == nil:
+				c.nsWhitelistLogged[name] = false // reset, so if the namespace goes away we'll log it again
+				nsList = append(nsList, *ns)
+			case apierrors.IsUnauthorized(err) || apierrors.IsForbidden(err) || apierrors.IsNotFound(err):
+				if !c.nsWhitelistLogged[name] {
+					c.logger.Log("warning", "whitelisted namespace unauthorized, forbidden, or not found", "namespace", name)
```

Comment:
It would be rather nice to know _which_ of the three errors it is. Why not simply log `err`? AFAICT (from looking at https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go) that should produce something reasonably intelligible.

Reply:
I suppose one issue here is that the error may change. If we care enough, we could stash the last error message (well, ideally code, but getting hold of that requires some casting) in the map.

Reply:
I changed it to log the error, so that will give people an indication of what's wrong. Logging transitions between the different errors is overkill I reckon.